lib/extras/codec_jpg.cc

```diff
@@ -195,11 +194,11 @@ void WriteExif(jpeg_compress_struct* const cinfo, const PaddedBytes& exif) {
   if (exif.size() < 4) return;
   jpeg_write_m_header(
       cinfo, kExifMarker,
-      static_cast<unsigned int>(exif.size() - 4 + sizeof kExifSignature));
+      static_cast<unsigned int>(exif.size() + sizeof kExifSignature));
   for (const unsigned char c : kExifSignature) {
     jpeg_write_m_byte(cinfo, c);
   }
-  for (size_t i = 4; i < exif.size(); ++i) {
+  for (size_t i = 0; i < exif.size(); ++i) {
```

Comment:
why this change from skipping 4 bytes to not skipping them?

Reply:
I think it's a bug caused by confusion about whether the blob starts immediately with the tiff header, or it starts with a 4-byte offset (that is always zero) and then you get the tiff header.

In the file format, the `Exif` box does do the offset thing (because that's how it's defined in ISOBMFF and done by others like jpeg 2000). In the blobs we are passing here, we don't do the offset thing (we have an implicit offset 0). In jpeg and png, there's also no offset.